libbeat/cfgfile/reload.go

```diff
@@ -15,3 +31,163 @@ type ReloadConfig struct {
 	Period  time.Duration `config:"period"`
 	Enabled bool          `config:"enabled"`
 }
+
+type RunnerFactory interface {
+	Create(*common.Config) (Runner, error)
+}
+type Runner interface {
+	Start()
+	Stop()
+	ID() uint64
```

Comment:
I was hoping yes, but heartbeat will be the next proving ground which could affect this one again. I would expect heartbeat to use the same hashing on the config options to generate the hash.